### PR TITLE
Add test plan for QUARKUS-451: Reactive Kafka

### PR DESCRIPTION
**Pull request by pjgg**
QUARKUS-451 - Testing and verifying OpenShift Kafka with Quarkus

### CODE REVIEW

File: QUARKUS-451.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# QUARKUS-451 - Testing and verifying OpenShift Kafka with Quarkus
+
+The goal of this test plan is to cover the main points of conflicts between Quarkus and Kafka. 
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-451
```
Comment (rsvoboda) [inline QUARKUS-451.md:5]: This link is not relevant, there is no good candidate atm, please drop the line

File: QUARKUS-451.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Scope of the testing
+Following actions were taken to ensure test coverage and automation for reactive kafka applications:
+ - Kafka producer with JSON / AVRO body messages.
+ - Kafka consumer with JSON / AVRO body messages.
+ - Kafka Streams with JSON body messages and windowed.
+ - Was tested over Kafka Strimzi and RH AMQ Streams.
```
Comment (rsvoboda) [inline QUARKUS-451.md:12]:
```suggestion
 - Tested over Kafka Strimzi and RH AMQ Streams.
```

File: QUARKUS-451.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ - Platform Openshift, Java 11.
```
Comment (rsvoboda) [inline QUARKUS-451.md:13]:
```suggestion
 - Platform OpenShift, Java 11.
```

File: QUARKUS-451.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+### Impact on testsuites and testing automation:
+ - OpenShift TS framework needs to be enhanced to support OpenShift kafka
```
Comment (rsvoboda) [inline QUARKUS-451.md:16]:
```suggestion
 - OpenShift TS framework needs to be enhanced to support Kafka on OpenShift
```

File: QUARKUS-451.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+   - Would be great if in the future we support some kind of openshift/k8s templating. So install some specifict Kafka (provider/version) would be as simple as apply a template.
```
Comment (rsvoboda) [inline QUARKUS-451.md:17]: Please create issue for this in OpenShiftTS

File: QUARKUS-451.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ - Kafka Streams with JSON body messages and windowed.
+ - Was tested over Kafka Strimzi and RH AMQ Streams.
+ - Platform Openshift, Java 11.
+
+### Impact on testsuites and testing automation:
+ - OpenShift TS framework needs to be enhanced to support OpenShift kafka
+   - Would be great if in the future we support some kind of openshift/k8s templating. So install some specifict Kafka (provider/version) would be as simple as apply a template. 
+
+### Impact on resources:
+ - More services will be running on OpenShift cluster, impact on CPU/memory should be minimal (if you don't setup Kafka/zookeper/registry with a persistent volume).
```
Comment (rsvoboda) [inline QUARKUS-451.md:20]: How many new services will be running with Kafka in the scope?

Kafka is perceived as lightweight but the reality is not so optimistic.

File: QUARKUS-451.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+- [Kafka AVRO Scenario](https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/140): There is an EventsProducer that generates stock price events every 1s. The events are typed by an AVRO schema.  
+A Kafka consumer will read these events serialized by AVRO and change a `status` property to `COMPLETED`. 
+The streams of completed events will be exposed through an SSE endpoint.
+
+## Advanced topics for test development
+The following advanced topics are listed for future consideration. There should be dedicated feature requirements and/or enhancements for these specific areas.
+ - Security testing
+ - Recovery/Failover testing
+ - Performance/Scalability testing
+ - Backward compatibility testing
```
Comment (rsvoboda) [inline QUARKUS-451.md:41]: What did you have in mind? Backward compatibility of clients or brokers? Should that be concern of Quarkus testing?

**Review by rsvoboda — CHANGES_REQUESTED** [review]

File: QUARKUS-451.md (hunk 6 above)
Comment (pjgg) [reply QUARKUS-451.md:20]: Well, that is true, Kafka is not "lightweight":

- 1 Kafka broker (without replication) (755.5 MiB - 343.5 MiB)
- 1 zookeeper () (61.8 MiB)
- 1 Kafka registry (128Mb)

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**